specs/darwin/keychain_items.table

```diff
@@ -8,6 +8,7 @@ schema([
     Column("created", TEXT, "Date item was created"),
     Column("modified", TEXT, "Date of last modification"),
     Column("type", TEXT, "Keychain item type (class)"),
+    Column("pk_hash", TEXT, "Hash of associated public key (SHA1 of DER PKCS#1)"),
```

Comment:
Is this a standard hash? Can we match openssl's format for it?

Reply:
It seems like it is actually the SHA1 hash of the `subjectPublicKey` bit string defined by RFC 5280, which matches the first method outlined in section [4.2.1.2](https://www.rfc-editor.org/rfc/rfc5280#section-4.2.1.2) for generating a "Subject Key Identifier". I'll update the documentation to reference that RFC instead of PKCS#1, since PKCS#1 is specifically for RSA keys.

I didn't find a particularly convenient way to output this in OpenSSL, though I might have missed it. One method I used to test is to first find the public key info section with the BIT STRING in the output of `openssl asn1parse -i`, which looks something like this:
```
  # ECDSA Example
  267:d=3  hl=2 l=  19 cons: SEQUENCE          
  269:d=4  hl=2 l=   7 prim: OBJECT            :id-ecPublicKey
  278:d=4  hl=2 l=   8 prim: OBJECT            :prime256v1
  288:d=3  hl=2 l=  66 prim: BIT STRING        

  # RSA Example
  278:d=2  hl=4 l= 290 cons:   SEQUENCE          
  282:d=3  hl=2 l=  13 cons:    SEQUENCE          
  284:d=4  hl=2 l=   9 prim:     OBJECT            :rsaEncryption
  295:d=4  hl=2 l=   0 prim:     NULL              
  297:d=3  hl=4 l= 271 prim:    BIT STRING        
```

The first number in the row is the byte offset into the ASN.1 DER encoded certificate data, `hl` is the header length of the section, and `l` is the length of the section content. The actual data to hash is offset by 1 byte within the BIT STRING. So the data to hash starts at `(offset + header + 1)` for `length - 1` bytes.

For the certificate examples above, the `pk_hash` can then be computed like:
```
# ECDSA example
$ cat ecdsa.pem | sed '1d;$d' | base64 -d | dd skip=291 bs=1 count=65 | sha1sum
# RSA example
$ cat rsa.pem | sed '1d;$d' | base64 -d | dd skip=302 bs=1 count=270 | sha1sum
```

Reply:
Works for me! Mostly I'm never sure when hash based fingerprints should be upcased or downcased of need `:` in the middle. Though I guess we can update if we figure out we got it wrong